frontend/src/App.tsx

```diff
@@ -10,49 +10,67 @@ import { CalendarProvider } from 'contexts/calendarContext';
 import useQuery from 'hooks/useQuery';
 import { getLoginUserDataApi, accessTokenRefreshApi } from 'apis/userApis';
 import useInterceptor from 'hooks/useInterceptor';
+import { AxiosRequestConfig } from 'axios';
 import { TokenStatus } from 'types/userType';
-import { queryClient } from 'contexts/queryClient';
+import { publicRequest, privateRequest } from 'apis/instances';
 import * as S from './App.styled';
 
 const App = () => {
   const userState = useContext(userContext) as UserContextValues;
 
   useInterceptor({
-    accessToken: userState.accessToken,
-    onRequest: (url, options, accessToken) => ({
-      url,
-      options: {
-        ...options,
-        headers: {
-          ...options.headers,
-          Authorization: `Bearer ${accessToken}`,
-        },
+    onRequest: (config: AxiosRequestConfig<unknown>) => ({
+      ...config,
+      headers: {
+        ...config.headers,
+        Authorization: `Bearer ${userState.accessToken}`,
       },
     }),
-    onError: (response, body) => {
-      if (response.status !== 401) {
-        return;
+    onRejected: async (error: any) => {
+      const {
+        config,
+        response: {
+          status,
+          data: { tokenStatus },
+        },
+      } = error;
+
+      if (status !== 401) {
+        return Promise.reject(error);
       }
 
-      if (body.tokenStatus === TokenStatus['invalid']) {
+      if (tokenStatus === TokenStatus['invalid']) {
         userState.logout();
         return;
       }
 
-      if (body.tokenStatus === TokenStatus['expired']) {
-        refreshQuery.refetch();
-        return;
+      if (tokenStatus === TokenStatus['expired']) {
+        const {
+          data: { accessToken },
+        } = await publicRequest('/token/refresh');
+
+        const newConfig = {
+          ...config,
+          Authorization: `Bearer ${accessToken}`,
+        };
+
+        userState.setAccessToken(accessToken);
+
+        return privateRequest(newConfig);
       }
 
       userState.setInitialized(true);
+      return Promise.reject(error);
     },
+    resolver: [userState],
```

Comment:
콜백 함수 내에서 사용되는 변수만 전달하도록 해야하지 않을까싶네요.  로그인, 로그아웃할 때 accessToken이 아니라 다른 상태 업데이트 작업이 있어서 로그인 한번에 effect 훅이 4번 실행됩니다. 

```suggestion
    resolver: [userState.accessToken],
```

Reply:
userState의 변경이 잦아서 불필요한 실행이 반복되고 있었네요. 사용하는 accessToken만 넘겨주는 것으로 수정했습니다. 👍